backend/data/src/test/kotlin/io/tolgee/unit/formats/apple/in/XcstringsFormatProcessorTest.kt

```diff
@@ -0,0 +1,129 @@
+package io.tolgee.unit.formats.apple.`in`
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import io.tolgee.formats.apple.`in`.xcstrings.XcstringsFileProcessor
+import io.tolgee.testing.assert
+import io.tolgee.unit.formats.PlaceholderConversionTestHelper
+import io.tolgee.util.*
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class XcstringsFormatProcessorTest {
+    lateinit var mockUtil: FileProcessorContextMockUtil
+
+    @BeforeEach
+    fun setup() {
+        mockUtil = FileProcessorContextMockUtil()
+        mockUtil.mockIt("example.xcstrings", "src/test/resources/import/apple/example.xcstrings")
+    }
+
+    @Test
+    fun `returns correct parsed result`() {
+        processFile()
+        mockUtil.fileProcessorContext.assertLanguagesCount(2)  // en, fr
+        mockUtil.fileProcessorContext.assertTranslations("en", "hello-world")
+            .assertSingle {
+                hasText("Hello, World!")
+            }
+        mockUtil.fileProcessorContext.assertTranslations("fr", "hello-world")
+            .assertSingle {
+                hasText("Bonjour le monde!")
+            }
+    }
+
+    @Test
+    fun `handles plural translations correctly`() {
+        processFile()
+        mockUtil.fileProcessorContext.assertTranslations("en", "messages-count")
+            .assertSinglePlural {
+                hasText(
+                    """
+                    {0, plural,
+                    one {You have # message}
+                    other {You have # messages}
+                    }
+                    """.trimIndent()
+                )
+                isPluralOptimized()
+            }
+    }
+
+    @Test
+    fun `import with placeholder conversion (disabled ICU)`() {
+        mockPlaceholderConversionTestFile(convertPlaceholders = false, projectIcuPlaceholdersEnabled = false)
+        processFile()
+        mockUtil.fileProcessorContext.assertLanguagesCount(1)
+        mockUtil.fileProcessorContext.assertTranslations("en", "messages-count")
+            .assertSinglePlural {
+                hasText(
+                    """
+                    {value, plural,
```

Comment:
Here we also need to test what happens when there is some ICU syntax by any chance in the string. Such syntax has to be escaped. 

This is tested here:
io.tolgee.unit.formats.apple.in.StringsdictFormatProcessorTest#import with placeholder conversion (disabled ICU)